File: Project3/python/task3.py
# task3.py
# Chandrachud Malali Gowda and Parth Parth
# This program is a sample in the selected language that demonstrates the rules for identifier naming, variable declarations, 
# and identifier scoping
# 2022 - 02 - 02

class MyTask3:

    # Constructor
    def __init__(self, name, age, height, weight):
        self.name = name # String data type
        self.age = age # Integer data type
        self.height = height # Float data type
        self.weight = weight # Float data type

    # Getter method for the name
    def get_name(self):
        return self.name

    # Setter method for the name
    def set_name(self, name):
        self.name = name

    # Getter method for the age
    def get_age(self):
        return self.age

    # Setter method for the age
    def set_age(self, age):
        self.age = age

    # Getter method for the height
    def get_height(self):
        return self.height

    # Setter method for the height
    def set_height(self, height):
        self.height = height

    # Getter method for the weight
    def get_weight(self):
        return self.weight

    # Setter method for the weight
    def set_weight(self, weight):
        self.weight = weight

    # Function to calculate and print the BMI
    def calculate_bmi(self):
        bmi = self.weight / (self.height * self.height)
        print("BMI is: ", bmi)

    # Function to reverse the age
    def reverse_age(self):
        rev = 0
        while(self.age > 0):
            rev = rev * 10 + self.age % 10
            self.age = self.age // 10 # // --> Floor division in python
        print("Reverse age is: ", rev)

    # Function to print the details of the person
    def print_details(self):
        print("Name: ", self.name)
        print("Age: ", self.age)
        print("Height: ", self.height)
        print("Weight: ", self.weight)

# Defining the main method
def main():
    task3 = MyTask3("Chandrachud", 22, 5.11, 70.5)
    task3.print_details()
    task3.calculate_bmi()
    task3.reverse_age()

if __name__ == '__main__':
    main()

File: Project3/r/task3.R
# task3.R
# Chandrachud Malali Gowda And Parth Parth
# This program is a sample python program that demosntrates all the
# built-in types and how to construct aggregate types, variable declarations,
# and identifier scoping
# 2022 - 02 - 02

library(methods)

# Creating the task3 reference class
my_task3 <- setRefClass("my_task3",
    fields = list(name = "character", age = "numeric", height = "numeric",
    weight = "numeric", alive = "logical"),
    methods = list(
    calculate_bmi = function(weight, height) {
        bmi <- 0
        bmi <- weight / (height * height)
        print(paste("The bmi is ", bmi))
    },
    reverse_age = function(age) {
        rev <- 0
        while (age > 0) {
            rev <- rev * 10 + age %% 10
            age <- floor(age / 10)
        }
        print(paste("The reversed age is ", rev))

    },
    print_details = function() {
        print(paste("Name: ", name))
        print(paste("Age: ", age))
        print(paste("Height: ", height))
        print(paste("Weight: ", weight))
        print(paste("Alive status: ", alive))
    }
))

task3 <- my_task3(name = "Chandrachud", age = 23, height = 1.75, weight = 65,
alive = TRUE)
task3$print_details()
task3$calculate_bmi(task3$weight, task3$height)
task3$reverse_age(task3$age)
